tests/common/include/tests/common/materials.inl

```diff
@@ -109,3 +111,33 @@ TEST(materials, mixture) {
     EXPECT_FLOAT_EQ(slab_vec[2].thickness_in_X0(),
                     slab3.thickness() / slab3.get_material().X0());
 }
+
+// This tests the material slab functionalities
+TEST(materials, material_slab) {
+
+    material_slab<scalar> slab(oxygen_gas<scalar>(), 2 * unit_constants::mm);
+
+    line_plane_intersection is;
+    is.cos_incidence_angle = scalar(0.3);
+
+    EXPECT_FLOAT_EQ(slab.interaction_length(is), 2 * unit_constants::mm / 0.3);
```

Comment:
Just as a comment: I just realized, this can potentially compare a ```float``` to a ```double```. Not terribly important right now, though

Reply:
I will just wrap them with scalar for now, but let's not get too strict about this right now.
unit_constants are defined with `double`, so we need to think about chaning it to struct and take scalar_type as template parameter

Reply:
Yes, I notices this, too. We should fix it at some point.